# old ppc archs are lacking the getauxval function and the two hardware capabilities
# that are added to cpu info.
# 
--- src/libFLAC/cpu.c.orig	2018-08-20 08:17:21.060232800 +0000
+++ src/libFLAC/cpu.c	2021-06-27 16:26:22.089868719 +0000
@@ -53,10 +53,6 @@
 #define dfprintf(file, format, ...)
 #endif
 
-#if defined FLAC__CPU_PPC
-#include <sys/auxv.h>
-#endif
-
 #if (defined FLAC__CPU_IA32 || defined FLAC__CPU_X86_64) && (defined FLAC__HAS_NASM || FLAC__HAS_X86INTRIN) && !defined FLAC__NO_ASM
 
 /* these are flags in EDX of CPUID AX=00000001 */
@@ -236,24 +232,9 @@
 static void
 ppc_cpu_info (FLAC__CPUInfo *info)
 {
-#if defined FLAC__CPU_PPC
-#ifndef PPC_FEATURE2_ARCH_3_00
-#define PPC_FEATURE2_ARCH_3_00		0x00800000
-#endif
-
-#ifndef PPC_FEATURE2_ARCH_2_07
-#define PPC_FEATURE2_ARCH_2_07		0x80000000
-#endif
-
-	if (getauxval(AT_HWCAP2) & PPC_FEATURE2_ARCH_3_00) {
-		info->ppc.arch_3_00 = true;
-	} else if (getauxval(AT_HWCAP2) & PPC_FEATURE2_ARCH_2_07) {
-		info->ppc.arch_2_07 = true;
-	}
-#else
+  // patched for ppc853x: none of those capabilities are available.
 	info->ppc.arch_2_07 = false;
 	info->ppc.arch_3_00 = false;
-#endif
 }
 
 void FLAC__cpu_info (FLAC__CPUInfo *info)
